Sources/ChessKitEngine/Engine.swift

```diff
@@ -39,72 +39,95 @@ public class Engine {
     ///
     public init(type: EngineType) {
         self.type = type
-        
         messenger = EngineMessenger(engineType: type.objc)
-        
-        messenger.responseHandler = { [weak self] response in
-            guard let self else { return }
-            
-            DispatchQueue.main.async {
-                if let parsedResponse = EngineResponse(rawValue: response) {
-                    self.log(parsedResponse.rawValue)
-                    self.receiveResponse(parsedResponse)
-                } else if !response.isEmpty {
-                    self.log(response)
-                }
-            }
-        }
     }
-    
+
     deinit {
         stop()
     }
-    
+
     /// Starts the engine.
     ///
     /// - parameter coreCount: The number of processor cores to use for engine
     /// calculation. The default value is `nil` which uses the number of
     /// cores available on the device.
+    /// - parameter multipv: The number of lines the engine should return,
+    /// sent via the `"MultiPV"` UCI option.
+    /// - parameter completion: The completion handler that is called when
+    /// the engine set up is complete. You must wait for this to be called
+    /// before sending further commands to the engine.
     ///
     /// This must be called before sending any commands
     /// with `send(command:)`.
-    public func start(coreCount: Int? = nil, multipv: Int = 1) {
+    public func start(coreCount: Int? = nil, multipv: Int = 1, completion: @escaping () -> Void) {
+        messenger.responseHandler = { [weak self] response in
+            guard let self else { return }
+
+            Task { @MainActor in
```

Comment:
One point I just stumbled upon: When activating StrictConcurrency warnings in this project this code introduces some warnings due to Engine not being Sendable and accessing self in a sendable closure (the Task closure). One might want to have a look at either making Engine Sendable, making Engine an actor which automatically marks it sendable or finding different solutions for these warnings.

Reply:
All good points, I need to revisit the Engine concurrency since it's a bit all over the place so I will take these comments into account before merging.